### DISCUSSION TITLE
Concerning network-manager-gnome on a server with two nics

### DISCUSSION THREAD

**Discussion by rkwesk** [ltsp/ltsp#174] [Q&A]
I am opening this as an issue so that anyone who has an opinion may comment/critisize on what I think is happenning so we all gain a better understanding.

I assume many are using network manager (not only the Ubuntu users) and so I usually test ltsp19 having commented out some lines in /etc/network/interfaces and instead use nm-connetion-editor and choose to edit the lan interface as a simple manual 192.168.67.1 without adding dns entries. As I understand it dnsmasq is handling all the dns needed for ltsp19. Also, Alkis has already explained that network manager must not be used to "share with other computers" the lan interface as it invokes another dnsmasq that conflicts with the one ltsp uses.

In contrast it was network manager that handled any specific dns servers in ltsp5 and then 127.0.0.1 was the ip entered that corresponded to dnsmasq as a dns cache.

Now that specific dns servers are listed in dnsmasq.conf, 0.0.0.0 is the ip that corresponds to dnsmasq as a dns cache.

Richard

**Comment by alkisg** [comment]
Richard I think there's indeed some confusion there. Let's try to clear it up.

You're correct to state that Ubuntu and Debian desktop editions use network-manager by default, so /etc/network/interfaces isn't the "default setup". This was the same in LTSP5.

> a simple manual 192.168.67.1 without adding dns entries

Let's say that you added 8.8.8.8 there. What would that mean? That your computer should ask 8.8.8.8 through that internal NIC. Can it? No. There are no DNS servers in the internal network that your LTSP server can access. So there's no point in adding DNS servers there.

So you're correct to say that dns entries make no sense in the internal connection. This was the same in LTSP5.

> dnsmasq is handling all the dns needed for ltsp19

By default, LTSP doesn't configure dnsmasq as a DNS server. On the other hand, sch-scripts does configure dnsmasq as a DNS server. Additionally, one can force it by passing a parameter to `ltsp dnsmasq`. This was the same in LTSP5.

> 127.0.0.1 was the ip entered that corresponded to dnsmasq as a dns cache.
> 0.0.0.0 is the ip that corresponds to dnsmasq as a dns cache.

These *only* make sense when someone diverts from the LTSP defaults and decides to activate DNS. Again, DNS=1 is NOT the default for LTSP, it's only the default for sch-scripts. In that case, 127.0.0.1 is what we put to network manager so that the server itself uses the dnsmasq cache. This was the same in LTSP5.

While 0.0.0.0 is that we put to dnsmasq.conf; then dnsmasq translates that to the "dnsmasq server IP", and sends it to the clients. This is new in LTSP19; previously, LTSP5 had some code that tried to autodetect if the LTSP server was also a DNS server or not; but since I read the dnsmasq.conf man page about 0.0.0.0, I decided that it's best to be explicit than to use autodetection/heuristics.
This is the only difference between LTSP5 and LTSP19.

Hope these helped a bit. :)

**Comment by rkwesk** [comment]
Oh, thank you for all this. I will read through it a few times before being ready to respond.

Indeed, I have much to learn.

Richard

**Comment by rkwesk** [comment]
Yes, I realized that dns=1 is not the default. I agree that the user should have the option of keeping it inactive or deliberately activating it.

I think that for schools having it activated is an advantage and so I add dns=1.

Also, from what you say here I realize that by putting the DNS_SERVER=0.0.0.0 (followed by one or two external servers) variable in the (common) section in ltsp.conf the clients will use the dns cache on the server as their first choice as well as the server. This is imho the best scenario for the schools and I thank you for explaining it.

Richard

**Comment by alkisg** [comment]
sch-scripts configure ltsp to the way that I think is best for schools.
Have a look at ltsp.conf that sch-scripts uses:
https://gitlab.com/sch-scripts/sch-scripts/blob/master/share/sch-scripts/ltsp.conf

Notice the DNS=1 there. It's under [common]. The server picks it up on `ltsp dnsmasq`, and the clients pick it up when they boot and use the LTSP server IP first.
We want schools to use DNS=1 because this way they can blacklist facebook etc, while it might not be a good default for businesses, that's why DNS=1 isn't the default for upstream LTSP.

So, normally, there's no need to use DNS_SERVER in ltsp.conf at all, neither in schools nor in other setups.

**Comment by rkwesk** [comment]
Thank you fir showing me this. May I use this with ltsp19 in the Greek schools? If so, how should add it?

Richard

**Comment by alkisg** [comment]
You just need to run this after sch-scripts gets installed:

```shell
sudo /usr/share/sch-scripts/initial-setup.sh
```

Greek schools are supported in the forum though, so let's avoid github to avoid confusing all the other LTSP users.

You may ask whatever you want about LTSP19 / sch-scripts / Greek schools there:
http://alkisg.mysch.gr/steki/index.php?topic=7858.0
